### PR TITLE
ui#482 - Statement - transfers aren't shown properly

### PR DESCRIPTION
**Pull request by ChriJ**

### CODE REVIEW

File: model/BeneficiaryTransferDestinationResponse.ts [hunk 1]
```diff
@@ -0,0 +1,12 @@
...
+	name: string
+}
+
+export namespace BeneficiaryTransferDestinationResponse {
+	export function is(
+		value: BeneficiaryTransferDestinationResponse | any
+	): value is BeneficiaryTransferDestinationResponse {
+		return typeof value == "object" && typeof value.beneficiaryId == "string" && typeof value.name == "string"
+	}
+}
```
Comment (lucy-meachem) [inline model/BeneficiaryTransferDestinationResponse.ts]: this one's wrong, it extends the AccountDetails one

File: model/StatementTransferSpecificType.ts [hunk 2]
```diff
@@ -0,0 +1,17 @@
+const statementTransferSpecificType = [
```
Comment (lucy-meachem) [inline model/StatementTransferSpecificType.ts:1]: missed REV

File: model/TransferResponseV2.ts [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+			(value.errorMessage == undefined || typeof value.errorMessage == "string") &&
+			Date.is(value.createdDate) &&
+			(value.paymentDate == undefined || Date.is(value.paymentDate)) &&
+			typeof value.reference == "string" &&
+			FundingAccountResponseV2.is(value.source) &&
+			TransferDestinationResponse.is(value.destination) &&
+			(value.bookingInfo == undefined || BookingInfoResponse.is(value.bookingInfo)) &&
+			typeof value.createdBy == "string"
+		)
+	}
```
Comment (lucy-meachem) [inline model/TransferResponseV2.ts:42]: this is wrong for the non-paxpay transfers

**Review by lucy-meachem — COMMENTED** [review]
needs unit tests

File: model/TransferResponseV2.ts (hunk 3 above)
Comment (ChriJ) [reply model/TransferResponseV2.ts:42]: made a summary type

File: model/BeneficiaryTransferDestinationResponse.ts (hunk 1 above)
Comment (ChriJ) [reply model/BeneficiaryTransferDestinationResponse.ts]: tried to emulate the extend bit in is function

File: model/BeneficiaryTransferDestinationResponse.ts [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+			typeof value == "object" &&
+			(value.sortCode == undefined || typeof value.sortCode == "string") &&
+			(value.accountNumber == undefined || typeof value.accountNumber == "string") &&
+			(value.iban == undefined || typeof value.iban == "string") &&
+			(value.bic == undefined || typeof value.bic == "string") &&
+			(value.fullName == undefined || typeof value.fullName == "string") &&
+			typeof value.beneficiaryId == "string" &&
+			(value.name == undefined || typeof value.name == "string")
+		)
+	}
```
Comment (lucy-meachem) [inline model/BeneficiaryTransferDestinationResponse.ts]: why not call AccountDetailsTransferDestinationResponse.is(this) && etc

Comment (lucy-meachem) [reply model/BeneficiaryTransferDestinationResponse.ts]: value not this

Comment (ChriJ) [reply model/BeneficiaryTransferDestinationResponse.ts]: the type system then thinks it has he first type for the following checks, e.g. beneficiaryId and says that is does not exist on that type

**Review by lucy-meachem — APPROVED** [review]

**Pull request merged**